### PR TITLE
Enhancement/query on param

### LINKED ISSUE #90: Expand posibilities to obtain tournaments
What has to be done:
Add options for obtaining tournaments. For example. Add option so that tournaments can be obtained using their name.

Why it has to be done:
So that the client can find the tournaments easier.

When the task is done:
When the client can make a request with multiple search options.

### PR DESCRIPTION
**Pull request by lfb0801**
closes #90 for now, more params would have to be added later

### CODE REVIEW

File: src/test/java/com/fontys/api/mockrepositories/MockTournamentRepository.java [hunk 1]
```diff
@@ -3,6 +3,12 @@
 import com.fontys.api.entities.Tournament;
 import com.fontys.api.repositories.TournamentRepository;
```
Comment (larsvandriel) [inline src/test/java/com/fontys/api/mockrepositories/MockTournamentRepository.java]: This file will be deleted in future release. You should merge with the master.

File: src/main/java/com/fontys/api/repositories/TournamentRepository.java [hunk 2]
```diff
@@ -3,6 +3,9 @@
 import com.fontys.api.entities.Tournament;
 import org.springframework.data.jpa.repository.JpaRepository;
 
+import java.util.Optional;
+
 public interface TournamentRepository extends JpaRepository<Tournament, Integer>
 {
+    Optional<Tournament> findByName(String name);
```
Comment (larsvandriel) [inline src/main/java/com/fontys/api/repositories/TournamentRepository.java:12]: Shouldn't this method have an implementation somewhere?

Comment (larsvandriel) [inline src/main/java/com/fontys/api/repositories/TournamentRepository.java:12]: Shouldn't this method have an implementation somewhere?

**Review by larsvandriel — CHANGES_REQUESTED** [review]

File: src/main/java/com/fontys/api/repositories/TournamentRepository.java (hunk 2 above)
Comment (lfb0801) [reply src/main/java/com/fontys/api/repositories/TournamentRepository.java:12]: Nope, let's thank the magic of hibernate

File: src/test/java/com/fontys/api/mockrepositories/MockTournamentRepository.java (hunk 1 above)
Comment (larsvandriel) [reply src/test/java/com/fontys/api/mockrepositories/MockTournamentRepository.java]: We discussed it and I approve.

File: src/main/java/com/fontys/api/repositories/TournamentRepository.java (hunk 2 above)
Comment (lfb0801) [reply src/main/java/com/fontys/api/repositories/TournamentRepository.java:12]: Nope, let's thank the magic of hibernate

Comment (larsvandriel) [reply src/main/java/com/fontys/api/repositories/TournamentRepository.java:12]: We discussed it and I approve.

Comment (larsvandriel) [reply src/main/java/com/fontys/api/repositories/TournamentRepository.java:12]: We discussed it and I approve.

**Review by MichielLangen — APPROVED** [review]

**Review by larsvandriel — APPROVED** [review]

**Pull request merged**